script.js

```diff
@@ -0,0 +1,337 @@
+  main.className = 'main';
+  const maincard = document.createElement('div');
+  maincard.className = 'maincard';
+  maincard.innerHTML = `
+
+  <button class="close-button2">&times;</button>
+  <div>
+ 
+  <img  class = "popup-image"src="${array[0].mobileImg}" alt="">
+  <img  class = "popup-image2"src="${array[0].desktopImg}" alt="">
+ </div>
+ <div class="headline">
+    <h2 class="popup-par">${array[0].name}</h2>
+    <div class ="frame18">
+       <button class="first-button2" type="button" id="messagep">See live <img src="${array[0].liveImg}" alt=""></button>
+       <button class="first-button1" type="button" id="messagep">See source <img src="${array[0].githubImg}" alt=""></button>
+    </div>
+  </div>   
+    <ul class="popup-list">
+      <li class="ruby">${array[0].tech[0]}</li>
+      <li class="css">${array[0].tech[1]}</li>
+      <li class="javascript">${array[0].tech[2]}</li>
+      <li class="html">${array[0].tech[3]}</li>
+      <li class="html">${array[0].tech[4]}</li>
+      <li class="html">${array[0].tech[5]}</li>
+    </ul>
+    <div class ="lorem">${array[0].desc}</div>
+    <div class ="frame19"><button class="mobile-first-button1" type="button" id="messagep">See source <img class="btn-img" src="${array[0].githubImg}" alt=""></button>
+       <button class="mobile-first-button2" type="button" id="messagep">See live <img class="btn-img" src="${array[0].liveImg}" alt=""></button></div>
+    `;
+
+  main.appendChild(maincard);
+  body.appendChild(main);
+
+  const closeButton2 = document.querySelector('.close-button2');
+  closeButton2.addEventListener('click', () => {
+    main.removeChild(maincard);
+    body.removeChild(main);
+  });
+});
+
+const contactForm = document.getElementById('form-id');
+const contactEmail = document.getElementById('email');
+const contactError = document.getElementById('errormessage');
+
+contactForm.addEventListener('submit', (event) => {
+  if (contactEmail.value.toLowerCase() !== contactEmail.value) {
+    event.preventDefault();
+    contactError.innerText = 'please type your email in LowerCase ,form not submitted';
+    contactError.style.position = 'inherit';
+  }
+});
+// Preserve data in the browser
+
+const form = document.getElementById('form-id');
+const nameInput = document.getElementById('name');
+const message = document.getElementById('message');
+const emailInput = document.getElementById('email');
+
+form.addEventListener('input', () => {
+  const formInfo = {
+    name: nameInput.value,
+    email: emailInput.value,
+    message: message.value,
+  };
+  localStorage.setItem('formDetails', JSON.stringify(formInfo));
+});
+
+function displayInfo() {
+  const info = JSON.parse(localStorage.getItem('formDetails'));
+  if (info) {
+    nameInput.value = info.name;
+    emailInput.value = info.email;
+    message.value = info.message;
+  }
+}
+
+displayInfo();
+// myFunction();
+// myFunction2();
+// project();
+// closePopup();
+// popup();
```

Comment:
- [ ] **[OPTIONAL]** Great job on your project! I wanted to offer an optional suggestion: it would be beneficial to remove the commented-out code in order to enhance the cleanliness and professionalism of your project.